modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaCdsProductPricer.java

```diff
@@ -0,0 +1,743 @@
+    ArgChecker.notNull(referenceDate, "referenceDate");
+    ArgChecker.notNull(refData, "refData");
+    if (!cds.getProtectionEndDate().isAfter(ratesProvider.getValuationDate())) { //short cut already expired CDSs
+      return CurrencyAmount.of(cds.getCurrency(), 0d);
+    }
+    LocalDate stepinDate = cds.getStepinDateOffset().adjust(ratesProvider.getValuationDate(), refData);
+    LocalDate effectiveStartDate = cds.getEffectiveStartDate(stepinDate);
+    recoveryRate(cds, ratesProvider); // for validation
+    Pair<CreditDiscountFactors, LegalEntitySurvivalProbabilities> rates = reduceDiscountFactors(cds, ratesProvider);
+    double protectionFull =
+        protectionFull(cds, rates.getFirst(), rates.getSecond(), referenceDate, effectiveStartDate);
+
+    return CurrencyAmount.of(cds.getCurrency(), -cds.getBuySell().normalize(cds.getNotional()) * protectionFull);
+  }
+
+  /**
+   * Calculates the present value sensitivity of the product. 
+   * <p>
+   * The present value sensitivity of the product is the sensitivity of present value to the underlying curves.
+   * 
+   * @param cds  the product 
+   * @param ratesProvider  the rates provider
+   * @param referenceDate  the reference date
+   * @param refData  the reference data
+   * @return the present value sensitivity
+   */
+  public PointSensitivityBuilder presentValueSensitivity(
+      ResolvedCds cds,
+      CreditRatesProvider ratesProvider,
+      LocalDate referenceDate,
+      ReferenceData refData) {
+
+    ArgChecker.notNull(cds, "cds");
+    ArgChecker.notNull(ratesProvider, "ratesProvider");
+    ArgChecker.notNull(referenceDate, "referenceDate");
+    ArgChecker.notNull(refData, "refData");
+    if (!cds.getProtectionEndDate().isAfter(ratesProvider.getValuationDate())) { //short cut already expired CDSs
+      return PointSensitivityBuilder.none();
+    }
+    LocalDate stepinDate = cds.getStepinDateOffset().adjust(ratesProvider.getValuationDate(), refData);
+    LocalDate effectiveStartDate = cds.getEffectiveStartDate(stepinDate);
+    double recoveryRate = recoveryRate(cds, ratesProvider);
+    Pair<CreditDiscountFactors, LegalEntitySurvivalProbabilities> rates = reduceDiscountFactors(cds, ratesProvider);
+
+    double signedNotional = cds.getBuySell().normalize(cds.getNotional());
+    PointSensitivityBuilder protectionLegSensi =
+        protectionLegSensitivity(cds, rates.getFirst(), rates.getSecond(), referenceDate, effectiveStartDate, recoveryRate)
+            .multipliedBy(signedNotional);
+    PointSensitivityBuilder riskyAnnuitySensi = riskyAnnuitySensitivity(
+        cds, rates.getFirst(), rates.getSecond(), referenceDate, stepinDate, effectiveStartDate)
+            .multipliedBy(-cds.getFixedRate() * signedNotional);
+
+    return protectionLegSensi.combinedWith(riskyAnnuitySensi);
+  }
+
+  //-------------------------------------------------------------------------
+  private double protectionLeg(ResolvedCds cds,
```

Comment:
add a `//` style comment to each private method.

The `ResolvedCds` argument should be on the next line